snuba/datasets/errors_replacer.py

```diff
@@ -298,8 +300,57 @@ def _build_group_replacement(
     )
 
 
+def _build_event_set_filter(
+    message: Mapping[str, Any], state_name: ReplacerState
+) -> Optional[Tuple[List[str], List[str], MutableMapping[str, str]]]:
```

Comment:
I understand we want to avoid accidental mutability but I believe it may be fine for return types (otherwise i need to clone the dict for nothing)

Reply:
This makes sense to me. If the return argument is supposed to be mutable, then we should mark it as such.